.local/bin/scripts/projectCreate

```diff
@@ -1,100 +1,189 @@
-  mkdir debug
-  touch src/main.cc
-  git init
-  echo "cmake_minimum_required(VERSION 2.8.9)" >> CMakeLists.txt
-  echo "project($pname)" >> CMakeLists.txt
-  echo "include_directories(include)" >> CMakeLists.txt
-  echo 'file(GLOB_RECURSE SOURCES "src/*".cc)' >> CMakeLists.txt
-  echo "add_executable($exe_name \${SOURCES})" >> CMakeLists.txt
-  echo "install(TARGETS $exe_name DESTINATION /usr/bin)" >> CMakeLists.txt
-  echo "#!/bin/sh" >> build
-  echo 'if [[ -z $1 ]]; then' >> build
-    echo 'mkdir -p bin' >> build
-    echo 'cd bin' >> build
-    echo 'cmake ..' >> build
-    echo 'make' >> build
-  echo 'elif [[ "$1" == "install" ]]; then' >> build
-    echo 'mkdir -p bin' >> build
-    echo 'cd bin' >> build
-    echo 'cmake ..' >> build
-    echo 'sudo make install' >> build
-  echo 'elif [[ "$1" == "debug" ]]; then' >> build
-    echo 'mkdir -p debug' >> build
-    echo 'cd debug' >> build
-    echo 'cmake -DCMAKE_BUILD_TYPE=Debug ..' >> build
-    echo 'make' >> build
-  echo 'elif [[ "$1" == "project" ]]; then' >> build
-    echo 'mkdir -p bin' >> build
-    echo 'cd bin' >> build
-    echo 'cmake -DCMAKE_EXPORT_COMPILE_COMMANDS=1 ..' >> build
-    echo 'cp compile_commands.json ..' >> build
-  echo 'fi' >> build
-  chmod u+x build
-  echo 'bin/*' >> .gitignore
-  echo 'debug/*' >> .gitignore
-  echo 'compile_commands.json' >> .gitignore
-  echo '.vimspector.json' >> .gitignore
-  echo '.clangd/*' >> .gitignore
-  echo '{' >> .vimspector.json
-  echo '"configurations": {' >> .vimspector.json
-  echo '"Launch": {' >> .vimspector.json
-  echo '"adapter": "vscode-cpptools",' >> .vimspector.json
-  echo '"configuration": {' >> .vimspector.json
-  echo '"request": "launch",' >> .vimspector.json
-  echo "\"program\": \"debug/$exe_name\"," >> .vimspector.json
-  echo "\"cwd\": \"`pwd`\"," >> .vimspector.json
-  echo '"externalConsole": true,' >> .vimspector.json
-  echo '"MIMode": "gdb"' >> .vimspector.json
-  echo '}' >> .vimspector.json
-  echo '}' >> .vimspector.json
-  echo '}' >> .vimspector.json
-  echo '}' >> .vimspector.json
-  ./build project
-elif [ "$1" == "py" ] || [ "$1" = "python" ]; then
-  printf "Project name: "
-  read -r pname
-  mkdir $pname || exit
-  _pname=$(echo $pname | sed -e 's/-/_/g')
-  mkdir -p $pname/docs
-  mkdir -p $pname/$_pname
-  mkdir -p $pname/tests
-  touch $pname/LICENSE
-  touch $pname/README.md
-  touch $pname/TODO.md
-  touch $pname/setup.py
-  touch $pname/.gitignore
-  touch $pname/install.sh
-  touch $pname/$_pname/__init__.py
-  touch $pname/$_pname/utils.py
-  touch $pname/$_pname/__main__.py
-fi
+[[ -z $1 ]] && help
+
+git_init='false'
+
+while getopts "l:hgp:c:e:" flag; do
+    case "${flag}" in
+        g)
+            git_init='true'
+            ;;
+        l)
+            lang="${OPTARG}"
+            ;;
+        p)
+            project_name="${OPTARG}"
+            ;;
+        c)
+            class_name="${OPTARG}"
+            ;;
+        e)
+            exe_name="${OPTARG}"
```

Comment:
options like class_name and exe_name should not be provided as command-line argument as they are not generalised options and specific to some certain language. Instead we can take these input interactively for different languages (if it is needed). For example, lua/rust may have some other options required to build project then adding those to command line options would increase an overhead for users to learn those options and would be a little bit inconvenient.

Reply:
So you are suggesting to just take -g and -l options ?

Reply:
something like this 
  [[  -z $project_name ]] && printf "Enter project Name : " && read project_name
  [[  -z $class_name ]] && printf "Enter class Name : " && read class_name
(one small request , can you add hacktoberfest topic to your repo  so these prs will count in my hacktoberfest contribution)

Reply:
yeah exactly like that. -g, -l and -p are options that I think we should support. And for -l and -p we can also have interactive fallback as you mentioned in comment.

Reply:
check this out -> https://github.com/planetplut0/archrice/blob/master/.local/bin/scripts/projectCreate